server/auth.go

```diff
@@ -274,19 +291,68 @@ func register(ur *UserRegisterRequest, initialPerm int, db *gorm.DB) (AuthRespon
 }
 
 func registerFirstUser(user *UserRegisterRequest, db *gorm.DB) (AuthResponse, error) {
-	// Ensure no users exist
+	err := validateNoExistingUsers(db)
+	if err != nil {
+		return AuthResponse{}, err
+	}
+	slog.Info("Registering first user.")
+	return register(user, PERM_ADMIN, db)
+}
+
+func registerFirstPlexUser(token *PlexUserRequest, db *gorm.DB) (AuthResponse, error) {
+	err := validateNoExistingUsers(db)
+	if err != nil {
+		return AuthResponse{}, err
+	}
+	slog.Info("Fetching metadata from Plex.")
+	username, err := fetchPlexUsernameFromToken(token.AuthToken)
+	if err != nil {
+		slog.Error("registerFirstPlexUser: Could not fetch Plex username", "error", err)
+		return AuthResponse{}, errors.New("could not fetch plex username")
+	}
+	user := UserRegisterRequest{Username: username, Password: token.AuthToken, Type: PLEX_USER}
```

Comment:
Using the auth token as the password is just temporary. In the future, when pulling plex history, the token would need to be stored somewhere in plaintext, and the password should instead be a non-related random string.

Reply:
Yeah, I can take a look at doing it that way instead. I wasn't sure if the API's were still protected w/o the password filled out and I didn't get to reading the code for it just yet.

Reply:
That's awesome, thanks @stephaje!

Yep as long as we only give the user their watcharr token after we authenticate them against plex (and locate/add them to our db) then we're all good